ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -0,0 +1,358 @@
+import ManagedCard from "../lib/ManagedCard";
+import SeparatedItems from "../lib/SeparatedItems";
+import LinkButton from "../LinkButton";
+import LoadingButton from "../LoadingButton";
+import AddSampleRecordModal from "./AddSampleRecordModal";
+import ArrayRecordPropertyList from "./ArrayRecordPropertyList";
+
+interface Props {
+  execApi: ApiHook["execApi"];
+  properties: Models.PropertyType[];
+  disabled: boolean;
+}
+
+const isConfigUI = grouparooUiEdition() === "config";
+
+const getCachedSampleRecordId = (modelId: string): string => {
+  return globalThis.localStorage?.getItem(`sampleRecord:${modelId}`);
+};
+
+const setCachedSampleRecordId = (modelId: string, recordId: string): void => {
+  if (recordId) {
+    globalThis.localStorage?.setItem(`sampleRecord:${modelId}`, recordId);
+  }
+};
+
+const clearCachedSampleRecordId = (modelId: string): void => {
+  globalThis.localStorage?.removeItem(`sampleRecord:${modelId}`);
+};
+
+const SampleRecordCard: React.FC<Props> = ({
+  properties,
+  execApi,
+  disabled,
+}) => {
+  const model = useGrouparooModelContext();
+  const prevModelId = usePrevious(model.id);
+
+  const [loading, setLoading] = useState(false);
+  const [importing, setImporting] = useState(false);
+  const [addingRecord, setAddingRecord] = useState(false);
+  const [hasRecords, setHasRecords] = useState(true);
+  const [record, setRecord] = useState<Models.GrouparooRecordType>();
+  const [groups, setGroups] = useState<Models.GroupType[]>();
+  const [destinations, setDestinations] = useState<Models.DestinationType[]>();
+  const [recordId, setRecordId] = useState(() =>
+    getCachedSampleRecordId(model.id)
+  );
+
+  const saveRecord = useCallback(
+    (record, groups = undefined, destinations = undefined) => {
+      setLoading(true);
+      setRecord(record);
+      setGroups(groups);
+      setDestinations(destinations);
+      setRecordId(record.id);
+      setHasRecords(true);
+      setLoading(false);
+
+      setCachedSampleRecordId(model.id, record.id);
+    },
+    [model]
+  );
+
+  const loadRecord = useCallback(async () => {
+    setLoading(true);
+
+    let record: Models.GrouparooRecordType;
+    let groups: Models.GroupType[];
+    let destinations: Models.DestinationType[];
+
+    if (recordId) {
+      ({ record, groups, destinations } = await execApi<Actions.RecordView>(
+        "get",
+        `/record/${recordId}`
+      ));
+    } else {
+      ({ record, groups, destinations } = await execApi<Actions.RecordsList>(
+        "get",
+        "/records",
+        {
+          limit: 25,
+          offset: 0,
+          modelId: model.id,
+        }
+      ).then((response) => {
+        if (response.total === 0) {
+          setHasRecords(false);
+        } else {
+          const randomRecord = response?.records?.length
+            ? response.records[
+                Math.floor(Math.random() * response.records.length)
+              ]
+            : undefined;
```

Comment:
Calling out that for now the "random" button will re a random selection of the first 25 records